### ISSUE TITLE
Bar width not calculated effectively for "value" type axis

### ISSUE THREAD

**Issue by maliayas** [apache/incubator-echarts#11374]
### Version
4.3.0

### Reproduction link
[https://jsfiddle.net/1wvhf4x2/2/](https://jsfiddle.net/1wvhf4x2/2/)

### Steps to reproduce
Use "value" as x axis type and have big gaps in x axis values in the data. Example setup to reproduce:

```js
option = {
    "xAxis": {
        "show": true,
        "type": "value",
        "scale": true,
    },
    "yAxis": {
        "show": true,
    },
    "series": [{
        "type": "bar",
        "data": [
            ["2019", "600"],
            ["2020", "600"],
            ["2032", "600"],
        ]
    }]
};
```

### What is expected?
Bars should not be too wide and they must not overlap.

### What is actually happening?
Bars are too wide and they overlap.

---
This issue does not occur for ~~"time" and~~ "cateogry" axis types. In the above example, just change "value" to "category" and in the result, bar width is nicely adjusted. See https://jsfiddle.net/1wvhf4x2/3/

**Comment by Ovilia** [comment]
This may be the same or similar issue to #11145, which is scheduled to be fixed in v4.5.0.
I will leave this issue open and check when #11145 is fixed.

**Comment by maliayas** [comment]
Yes, it looks like they are the same. I'll follow both issues. Thanks very much.

PS: I was wrong in the OP when I said this doesn't happen with "time" axis. In fact this happens for both "time" and "value" axises. OP edited to reflect this.

**Comment by maliayas** [comment]
I can confirm this is fixed as of v4.6.0. Thanks very much.

I only have one question. Currently there may happen some overlapping with the first bar and the y axis. Do you think this should be fixed with better defaults or it's expected and I should use some other config (e.g. `xAxis.boundaryGap`) to fix it? JSFiddle: https://jsfiddle.net/bwez36hs/

**Comment by Ovilia** [comment]
@maliayas Sorry I just saw this. Please follow #11964.
